src/main/java/org/opensearch/knn/quantization/models/quantizationState/QuantizationStateCache.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationState;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.RemovalCause;
+import com.google.common.cache.RemovalNotification;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.common.unit.ByteSizeValue;
+import org.opensearch.knn.index.KNNSettings;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+import static org.opensearch.knn.index.KNNSettings.QUANTIZATION_STATE_CACHE_EXPIRY_TIME_MINUTES;
+import static org.opensearch.knn.index.KNNSettings.QUANTIZATION_STATE_CACHE_SIZE_LIMIT;
+
+/**
+ * A thread-safe singleton cache that contains quantization states.
+ */
+@Log4j2
+public class QuantizationStateCache {
+
+    private static volatile QuantizationStateCache instance;
+    private Cache<String, QuantizationState> cache;
+    @Getter
+    @Setter
+    private long maxCacheSizeInKB;
+    @Getter
+    private Instant evictedDueToSizeAt;
+
+    private QuantizationStateCache() {
+        maxCacheSizeInKB = ((ByteSizeValue) KNNSettings.state().getSettingValue(QUANTIZATION_STATE_CACHE_SIZE_LIMIT)).getKb();
+        buildCache();
+    }
+
+    /**
+     * Gets the singleton instance of the cache.
+     * @return QuantizationStateCache
+     */
+    public static QuantizationStateCache getInstance() {
```

Comment:
`QuantizationStateCache ` creates another cache layer in the code base for a specific purpose

This can be simplified to `Cache<String, QuantizationState> getInstance()`. methods like `getQuatizationState` are just wrappers over the exising Cache methods, with each functionality we add, we need another wrapper function. Would encourage you to rethink this so we don't have wrapper methods and this can be simply be used as initialize and tear down of the cache

Reply:
This is modeled after other similar caches in the plugin such as ModelCache and NativeMemoryCacheManager.  There is more functionality here, such as the expired time and max size in kb that will eventually be present in the KNNStats API, which will help users debug and have more insight into potential issues with their clusters.

Reply:
We'll take this as a follow-up PR after thinking about it more